src/server/bg_services/db_cleaner.js

```diff
@@ -58,23 +58,39 @@ async function clean_md_store(last_date_to_remove) {
         ${total_objects_count} objects - Skipping...`);
         return;
     }
+    const objects_to_remove = await clean_md_store_objects(last_date_to_remove, config.DB_CLEANER_DOCS_LIMIT);
+    const blocks_to_remove = await clean_md_store_blocks(last_date_to_remove, config.DB_CLEANER_DOCS_LIMIT);
+    const filtered_chunks = await clean_md_store_chunks(last_date_to_remove, config.DB_CLEANER_DOCS_LIMIT);
+
+    dbg.log0(`DB_CLEANER: removed ${objects_to_remove.length + blocks_to_remove.length + filtered_chunks.length} documents from md-store`);
+}
+
+async function clean_md_store_objects(last_date_to_remove, limit) {
     dbg.log0('DB_CLEANER: checking md-store for documents deleted before', new Date(last_date_to_remove));
-    const objects_to_remove = await MDStore.instance().find_deleted_objects(last_date_to_remove, config.DB_CLEANER_DOCS_LIMIT);
+    const objects_to_remove = await MDStore.instance().find_deleted_objects(last_date_to_remove, limit);
     dbg.log2('DB_CLEANER: list objects:', objects_to_remove);
-    if (objects_to_remove.length) {
+    if (objects_to_remove.length > config.MD_STORE_MAX_DELETED_OBJECTS_LIMIT) {
```

Comment:
why are we adding this limit? why not deleting less?

Reply:
What would be the good number ?

Reply:
I think number is arbitrary, but we can go with 10 or 20 for example